routing/router.go

```diff
@@ -1820,7 +1820,24 @@ func (r *ChannelRouter) sendPayment(
 // to continue with an alternative route. This is indicated by the boolean
 // return value.
 func (r *ChannelRouter) processSendError(paySession PaymentSession,
-	rt *route.Route, fErr *htlcswitch.ForwardingError) bool {
+	rt *route.Route, sendErr error) bool {
+
+	// If the failure message could not be decrypted, attribute the failure
+	// to our own outgoing channel.
+	//
+	// TODO(joostager): Penalize all channels in the route.
+	if sendErr == htlcswitch.ErrUnreadableFailureMessage {
```

Comment:
Yes. But that is currently already the case, as a decryption error is transformed into a self node failure.

In the follow up pr with proper failure attribution, this will be fixed.